src/app/components/elements/panels/UserPassword.tsx

```diff
@@ -17,10 +17,12 @@ interface UserPasswordProps {
     setNewPasswordConfirm: (e: any) => void;
     newPasswordConfirm: string;
     editingOtherUser: boolean;
+    doPasswordsMatch: boolean;
+    validPassword: boolean;
 }
 
 export const UserPassword = (
-    {currentPassword, currentUserEmail, setCurrentPassword, myUser, setMyUser, isNewPasswordConfirmed, userAuthSettings, setNewPassword, setNewPasswordConfirm, newPasswordConfirm, editingOtherUser}: UserPasswordProps) => {
+    {currentPassword, currentUserEmail, setCurrentPassword, myUser, setMyUser, isNewPasswordConfirmed, userAuthSettings, setNewPassword, setNewPasswordConfirm, newPasswordConfirm, editingOtherUser, doPasswordsMatch, validPassword}: UserPasswordProps) => {
 
```

Comment:
"isNewPasswordConfirmed, userAuthSettings, setNewPassword, setNewPasswordConfirm, newPasswordConfirm,..., doPasswordsMatch, validPassword"

There seem to be a lot of things here. Is this code starting to get a bit convoluted?

Reply:
It's not ideal, but since the form submission happens in the MyAccount page, the UserPassword needs access to current password and email, as well as the ability to set the current, new password & new password confirmation.

I added "doPasswordsMatch" and "validPassword" to those, because the props currently don't pass down the newPassword / newPasswordConfirm states but they are needed to run those PW checks. I could alternatively have a new state living inside UserPassword?

Reply:
maybe just tweak the naming a little.

ValidPassword - does that mean 'password that passes the password requirements'? or 'password that has been checked and is the right password'?

doPasswordsMatch - does that mean the new password and the confirmation field for the new password?

Reply:
I'll rename validPassword to passwordMeetsRequirements.

doPasswordsMatch is just comparing new password to new password confirm values.  I was trying to avoid a long name like newPasswordMatchesNewPasswordConfirmation (doesn't seem like an improvement!). arePasswordsEqual? passwordsAreMatching?

Reply:
I like 'arePasswords<something>' - 'arePasswordsIdentical' maybe, remove the word 'match' which might imply the 'password matching' process of login?